nested/network-vnet-noddos.json

```diff
@@ -0,0 +1,31 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "moodleCommon": {
+          "metadata": {
+            "description": "Common Moodle values"
+          },
+          "type": "object"
+        }
+    },
+    "resources": [
+      {
+        "type": "Microsoft.Network/virtualNetworks",
+        "apiVersion": "2018-02-01",
+        "location": "[resourceGroup().location]",
+        "name": "[parameters('moodleCommon').vnetName]",
+        "properties": {
+          "addressSpace": {
+            "addressPrefixes": [
+              "[concat(parameters('moodleCommon').vNetAddressSpace,'/16')]"
+            ]
+          }
+
+        }
+      }
+    ],
+    "variables": {
+        "documentation01": "This sub-template creates a virtual network when no customer BYO-vnet is specified"
+    }
+}
```

Comment:
Right, but then, since there's the boolean `enableDdosProtection` property anyway, I guess you can create a DDos protection plan no matter what, and just use that switch to enable/disable the plan? That way, it seems to me that you really don't need to have this hack at all. I'm assuming that the always-created DDos protection plan won't incur any charge as long as it's not enabled.

Still, this doesn't address the custom BYO-vnet issue. Eventually, you'll need to check if deploying the same/already created vnet again with additional properties is accepted by ARM, I think...

Reply:
If I set the enableDdosProtection property to true, I must set the ddosProtectionPlan Id on the vnet. If it is not set, a error will be raised.

I was thinking about your last proposition. Can I add a nested template that will only focus on setting up the DDoS protection plan and adding the required info on the vnet ?

Reply:
If I always create the DDoS protection plan resource, a 2,400$ fee will always be charged. I think it is not acceptable for someone who set the switch to false.

Reply:
I see. Thanks. So that's why I mentioned my assumption earlier: Created, but not enabled DDoS protection plan won't incur any charges. Are you really sure that just a created-but-not-enabled plan will still incur charges? That's pretty weird, I must say. If that's the case, yes, we shouldn't create a protection plan at all. However, I'd reconfirm that's really the case, and if so, why Azure charges a DDoS protection plan when it's not enabled, and what's the point of having the switch in the first place.

Reply:
OK, never mind what I said. I should have read the pricing documentation carefully. They do mention the fixed charge, so just creating a plan seems to incur that charge anyway. Sorry for not noticing that earlier, even though you mention that explicitly...

Reply:
I have reworked my solution. I am testing it. If I'm right, it will work on an existing vnet and a new one.